libs/dh/charges/feature-create-prices/src/lib/dh-charges-create-prices.component.ts

```diff
@@ -0,0 +1,358 @@
+    private router: Router
+  ) {}
+
+  ngOnInit() {
+    this.marketParticipantStore.loadMarketParticipants();
+
+    this.buildMarketParticipantOptions();
+    this.buildChargeTypeOptions();
+    this.buildValidityOptions();
+
+    this.chargesStore.createRequestHasError$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((hasError) => {
+        if (hasError) {
+          this.toastService.open({
+            message: this.translocoService.translate(
+              'charges.createPrices.createPriceError'
+            ),
+            type: 'danger',
+          });
+
+          this.enableFormGroup();
+        }
+      });
+
+    this.chargesStore.createRequestHasSucceeded$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((hasSucceded) => {
+        if (hasSucceded) {
+          this.toastService.open({
+            message: this.translocoService.translate(
+              'charges.createPrices.createPriceSuccess'
+            ),
+            type: 'success',
+          });
+
+          setTimeout(() => {
+            this.router.navigateByUrl(
+              `${dhChargesPath}/${dhChargesPricesPath}`
+            );
+          }, 1000);
+        }
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.unsubscribe();
+  }
+
+  ownerChanged(marketParticipantId: string) {
+    this.marketParticipantStore.all$
+      .pipe(take(1))
+      .subscribe((marketParticipants) => {
+        this.selectedSenderMarketParticipant = marketParticipants?.find(
+          (mp) => mp.id == marketParticipantId
+        );
+      });
+  }
+
+  chargeTypeChanged(chargeType: number) {
+    this.isTariff = chargeType == Number(ChargeTypes.Tariff);
+
+    // Building options because it's different depending on the chargetype.
+    this.buildValidityOptions();
+    this.enableFormGroup();
+
+    switch (Number(chargeType)) {
+      case ChargeTypes.Tariff: {
+        this.charge.controls['resolution'].setValue('');
+        this.charge.controls['transparentInvoicing'].setValue(true);
+        break;
+      }
+      case ChargeTypes.Subscription: {
+        this.charge.controls['transparentInvoicing'].setValue(true);
+        this.disableResolutionWithValue();
+        this.disableTaxIndicator();
+        break;
+      }
+      case ChargeTypes.Fee: {
+        this.disableTaxIndicator();
+        this.disableTransparentInvoicing();
+        this.disableResolutionWithValue();
+        break;
+      }
+      default:
+        break;
+    }
+  }
+
+  createPrice() {
+    if (!this.charge.valid) return;
+
+    this.disableFormGroup();
+    const charge = this.charge.value;
+
+    const resolution = this.charge.controls['resolution'].value;
+
+    this.chargesStore.createCharge({
+      chargeType: charge.chargeType as ChargeType,
+      effectiveDate: charge.effectiveDate as string,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      resolution: Resolution[Number(resolution)] as any,
```

Comment:
Because `CreateChargeV1Dto` expects the Resolution from the autogenerated enum model. Which we cannot import, because we are already importing `Resolution` from the Domain library. 

Maybe we could rename the `Resolution` from the Domain library to something else? Maybe ResolutionOptions?

Reply:
If `Resolution` in the Domain library is only used to dropdown options, then renaming it to `ResolutionOptions` is a good option.